Sources/MessagingPush/PushHandling/PushClickHandler.swift

```diff
@@ -16,17 +16,20 @@ protocol PushClickHandler: AutoMockable {
 class PushClickHandlerImpl: PushClickHandler {
     private let deepLinkUtil: DeepLinkUtil
     private let messagingPush: MessagingPushInstance
+    private let pushLogger: PushNotificationLogger
 
-    init(deepLinkUtil: DeepLinkUtil, messagingPush: MessagingPushInstance) {
+    init(deepLinkUtil: DeepLinkUtil, messagingPush: MessagingPushInstance, pushLogger: PushNotificationLogger) {
         self.deepLinkUtil = deepLinkUtil
         self.messagingPush = messagingPush
+        self.pushLogger = pushLogger
     }
 
     func trackPushMetrics(for push: PushNotification) {
         guard let cioDelivery = push.cioDelivery else {
+            pushLogger.logClickedPushMessageWithEmptyDeliveryId()
```

Comment:
just a callout: there is a warning that says this line is not being tested, are we tackling it in any other PR or any other reason?